web_services/foxx/brightid/encoding.js

```diff
@@ -127,6 +127,23 @@ function getConsensusSenderAddress() {
   return address;
 }
 
+function recoverEthSigner(signature, message) {
+  message = `\x19Ethereum Signed Message:\n${message.length}${message}`;
+  const hexMessage = Buffer.from(message, "binary").toString("hex");
+  const hashMessage = new Uint8Array(
+    createKeccakHash("keccak256").update(hexMessage, "hex").digest()
+  );
```

Comment:
This could be computed once per context and stored instead of computed every time a contextid is linked.